hyperspy/io_plugins/blockfile.py

```diff
@@ -287,7 +343,12 @@ def file_writer(filename, signal, **kwds):
                                         ('ID', endianess + 'u4')])
         dp_head['MAGIC'] = 0x55AA
         # Write by loop:
-        for img in signal._iterate_signal():
+        for img in tqdm(signal._iterate_signal(),
+                        total=signal.data.shape[0] * signal.data.shape[1]):
+            # to ensure dask datasets, get a tofile method
             dp_head.tofile(f)
+            img = np.array(img)
+            if scale_strategy is not None:
+                img = rescale_intensity(img, in_range=original_scale, out_range=np.uint8)
```

Comment:
Instead of using a `for` loop, would it be better to use `s.unfold_navigation_space` or `s.data.reshape(new_shape)`?

Reply:
I had a look at this and the performance when writing lazy signal and it is not very good, one order of magnitude worse than using hdf5 file - and there is faster than hdf5.
One issue is that it uses `LazySignal._iterate_signal`, which rechunk the data most likely unnecessarily. I think the best to fix this issue in a separate PR, which I would be happy to do!

Reply:
A loop never looks great in python and indeed performance is terrible, which is why I added the progress bar. I was thinking in the direction of using [`dask.array.store`](https://docs.dask.org/en/latest/array-api.html?highlight=store#dask.array.store) on a `np.memmap` object, but haven't experimented yet to see if it would work.

Reply:
so this seems to work and is in my opinion quite elegant. Probably it's also the fastest it will go. With the loop I was counting about an hour to convert one 11 GB lazy dataset into another, now it takes 3 minutes including the intensity rescaling. This involved the extraction from a `.hspy` file and saving to a `.blo` file. From what I could tell I did not have any issues with undue memory pressure, even though I saw this issue pop up https://github.com/dask/dask/issues/5985 . Perhaps something to keep in mind if there are any memory related issues in the future.